### PR TITLE
fix: validate ArnioCleaner parameters after set_params updates

### LINKED ISSUE #1660: Bug: validate ArnioCleaner parameters after set_params updates
## Summary

`ArnioCleaner` validates `copy` and `allow_row_count_change` in `__init__`, but scikit-learn users can change estimator parameters later through `set_params()`. Those updates bypass `__init__`, and `fit()` / `transform()` do not revalidate the parameters.

This allows invalid parameter values to affect transform behavior silently.

## Why it matters

Scikit-learn-compatible estimators are commonly configured through `Pipeline`, `GridSearchCV`, and `set_params()`. Arnio should validate parameters at use time as well as construction time so invalid configurations fail clearly.

## Evidence

- `arnio/integrations/sklearn.py`: `ArnioCleaner.__init__()` validates `copy` and `allow_row_count_change`.
- `arnio/integrations/sklearn.py`: `fit()` and `transform()` do not revalidate those attributes.
- `set_params()` is inherited from scikit-learn and updates attributes directly.

## Reproduction or trigger

```python
import pandas as pd
from arnio.integrations.sklearn import ArnioCleaner

cleaner = ArnioCleaner(steps=[])
cleaner.set_params(copy="not-a-bool")

out = cleaner.fit_transform(pd.DataFrame({"a": [1, 2]}))
print(out)
```

The invalid `copy` value is accepted and transform succeeds.

## Expected behavior

Invalid `copy` or `allow_row_count_change` values should raise a clear `TypeError` when the estimator is fitted or transformed, including when they are set through `set_params()`.

## Actual behavior

`set_params(copy="not-a-bool")` leaves the estimator in an invalid state and `fit_transform()` still succeeds.

## Suggested direction

Add a small internal parameter validation helper and call it from `fit()` and/or `transform()` before using the parameters. Add tests for:

- invalid `copy` set via `set_params()`
- invalid `allow_row_count_change` set via `set_params()`
- valid `set_params()` updates continuing to work
- normal sklearn `Pipeline` usage still passing

## References

- `arnio/integrations/sklearn.py::ArnioCleaner`

### PR DESCRIPTION
**Pull request by Aditya8369**
I added a `set_params()` override in sklearn.py that snapshots the current top-level parameters, applies the update, revalidates the runtime booleans, and rolls back to the previous state if validation or sklearn’s own parameter handling fails.

I also added regression coverage in test_sklearn.py, test_sklearn.py, and test_sklearn.py for invalid runtime updates, rollback after failure, valid updates returning self, and Pipeline delegation.

closes #1660

### CODE REVIEW

**Comment by Aditya8369** [comment]
@im-anishraj just a quick note, the checks in workflow are passing without any errors so you can merge it.
If there are changes needed from my side, I would be happy to do it.

**Review by im-anishraj — APPROVED** [review]
@Aditya8369 I reviewed the latest head `337e490` and re-ran the focused verification locally:

- `git diff --check origin/main...HEAD`: passed
- `python -m black --check arnio/integrations/sklearn.py tests/test_sklearn.py`: passed
- `python -m ruff check arnio/integrations/sklearn.py tests/test_sklearn.py`: passed
- `python -m pytest tests/test_sklearn.py -q`: 31 passed

I also added the matching GSSoC tracking labels to this PR. Approving from code review. The remaining Vercel failure is the fork authorization gate, not a code failure from these checks.

**Comment by im-anishraj** [comment]
@Aditya8369 final merge-side verification still passes on the latest head:

- git diff --check origin/main...HEAD -> passed
- black --check arnio/integrations/sklearn.py tests/test_sklearn.py -> passed
- ruff check arnio/integrations/sklearn.py tests/test_sklearn.py -> passed
- pytest tests/test_sklearn.py -q -> 31 passed

The GitHub matrix is green; the remaining Vercel status is only the external authorization gate.
